### PR TITLE
Add tools.go + setup k8s codegen in Makefile

### PR DESCRIPTION
**Pull request by epk**
## Description 

- Adding a `tools.go` to import and version codegen tooling. See https://github.com/go-modules-by-example/index/blob/master/010_tools/README.md
- Adding a make target to install `k8s.io/code-generator` tools (inline comments follow)

cc @carsonoid (since you last touched these files)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -276,34 +276,58 @@ minimaltools:
...
-	find $(GEN_BASE_DIR)/apis/topo/v1beta1 -type f -name 'zz_generated*' -exec rm '{}' \;
-	deepcopy-gen -i $(GEN_BASE_DIR)/apis/topo/v1beta1 -O zz_generated.deepcopy -o ./ --bounding-dirs $(GEN_BASE_DIR)/apis --go-header-file $(GEN_BASE_DIR)/boilerplate.go.txt
+	find $(VTROOT)/go/vt/topo/k8stopo/apis/topo/v1beta1 -name "zz_generated.deepcopy.go" -delete
 
-	# Delete, generate, and move the client libraries
-	rm -rf go/vt/topo/k8stopo/client
+	$(DEEPCOPY_GEN) --input-dirs $(GEN_BASE_DIR)/apis/topo/v1beta1 \
+	-O zz_generated.deepcopy \
+	--bounding-dirs $(GEN_BASE_DIR)/apis \
+	--go-header-file ./go/vt/topo/k8stopo/boilerplate.go.txt
```
Comment (epk) [inline Makefile:300]: `--go-header-file ./go/vt/topo/k8stopo/boilerplate.go.txt` this needs to be a relative path 

I ran into the following if I changed this flag to `--go-header-file  $(GEN_BASE_DIR)/boilerplate.go.txt`
```
F0102 14:59:42.633489   31170 deepcopy.go:131] Failed loading boilerplate: open vitess.io/vitess/go/vt/topo/k8stopo/boilerplate.go.txt: no such file or directory
```

File: Makefile [hunk 2]
```diff
@@ -276,34 +276,58 @@ minimaltools:
...
 
-	# There is no way to get client-gen to automatically put files in the right place and still have the right import path so we generate and move them
```
Comment (epk) [inline Makefile:290]: > There is no way to get client-gen to automatically put files in the right place and still have the right import path so we generate and move them

This is no longer true as newer versions of `k8s.io/code-generator` are Go modules aware BUT somehow they still rely on the repository being cloned to `$GOPATH/src/vitess.io/vitess` to put files in the right place

**Review by carsonoid — APPROVED** [review]

**Comment by carsonoid** [comment]
LGTM. Why the update though? Is there a CRD change in the works?

**Comment by epk** [comment]
> Is there a CRD change in the works?

I don't think so, I just want to remove an implicit tooling dependency and lock it's version at the same time

**Comment by epk** [comment]
@sougou let's get this merged?

**Comment by sougou** [comment]
Hi @epk. I had this on my TODO list (sorry it took so long, was busy with other things). Is there a reason why you closed this? Let me know if you want to reopen it.

**Comment by epk** [comment]
HI @sougou, sorry about that, if you can re-open the PR, I will bring it up to a merge-able state

**Comment by sougou** [comment]
Looks like you can't reopen. It says the source repo has been deleted.

**Comment by epk** [comment]
> Looks like you can't reopen. It says the source repo has been deleted.

I opened https://github.com/vitessio/vitess/pull/7517 with the same changes

**Pull request closed without merging**